src/main/java/edu/iis/powp/command/factory/CommandFactoryWindow.java

```diff
@@ -0,0 +1,268 @@
+
+
+    public CommandFactoryWindow(final CommandRegistry commandRegistry) {
+        this.commandRegistry = commandRegistry;
+        this.initialize();
+    }
+
+    private static final long serialVersionUID = -7773470008348070296L;
+    private JPanel freePanel = null;
+
+    private void initialize() {
+        this.setTitle("Command Factory");
+        this.setBounds(50, 100, 840, 520);
+        this.setDefaultCloseOperation(HIDE_ON_CLOSE);
+        this.setResizable(false);
+        GridBagLayout mainGridBagLayout = new GridBagLayout();
+        mainGridBagLayout.columnWidths = new int[]{0, 540};
+        mainGridBagLayout.rowHeights = new int[2];
+        mainGridBagLayout.columnWeights = new double[]{1.0D, 4.9E-324D};
+        mainGridBagLayout.rowWeights = new double[]{0.0D, 1.0D};
+        this.getContentPane().setLayout(mainGridBagLayout);
+        JPanel leftPanel = new JPanel();
+        GridBagLayout gridBagLayout = new GridBagLayout();
+        leftPanel.setLayout(gridBagLayout);
+        GridBagConstraints leftPanelConstraints = new GridBagConstraints();
+        leftPanelConstraints.fill = 1;
+        leftPanelConstraints.gridheight = 1;
+        leftPanelConstraints.gridx = 0;
+        leftPanelConstraints.gridy = 0;
+        leftPanelConstraints.weighty = 1;
+        this.getContentPane().add(leftPanel, leftPanelConstraints);
+
+        JList jList = new JList();
+        JScrollPane listScrollPane = new JScrollPane();
+        listScrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS);
+        listScrollPane.setViewportView(jList);
+
+        listScrollPane.setPreferredSize(new Dimension(0, 520/2));
+        DefaultListModel<IPlotterCommand> jListModel = new DefaultListModel<>();
+        jList.setModel(jListModel);
+        GridBagConstraints jListConstraints = new GridBagConstraints();
+        jListConstraints.fill = 1;
+        jListConstraints.gridheight = 1;
+        jListConstraints.gridx = 0;
+        jListConstraints.gridy = 1;
+        jListConstraints.weighty = 1;
+        this.getContentPane().add(listScrollPane, jListConstraints);
+
+
+        JComboBox<ConstructorDecorator> basicComboBox = new JComboBox<>();
+        basicComboBox.setPreferredSize(new Dimension(COMMON_WIDTH2, COMMON_HEIGHT));
+        for (Constructor<? extends IPlotterCommand> constructor : commandRegistry.getRegisteredBasicCommands()) {
+            basicComboBox.addItem(new ConstructorDecorator(constructor));
+        }
+        GridBagConstraints basicComboBoxConstraints = new GridBagConstraints();
+        basicComboBoxConstraints.gridx = 0;
+        basicComboBoxConstraints.gridy = 0;
+        leftPanel.add(basicComboBox,basicComboBoxConstraints);
+
+        JButton basicAddButton = new JButton("Add");
+        basicAddButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(final ActionEvent e) {
+                JFrame frame = new JFrame();
```

Comment:
FIX TO MERGE: Lack of separation between GUI and logic. Big class, hard to understand.